packages/frontend-2/components/settings/workspaces/automation/Functions/Functions.vue

```
@@ -0,0 +1,117 @@
+      ]"
+      :items="workspaceFunctions"
+    >
+      <template #name="{ item }">
+        <div class="flex items-center gap-2">
+          <AutomateFunctionLogo size="md" :logo="item.logo" />
+          <NuxtLink
+            class="text-foreground-3 hover:text-foreground-2 underline"
+            :to="automateFunctionRoute(item.id)"
+          >
+            {{ item.name }}
+          </NuxtLink>
+        </div>
+      </template>
+      <template #owner="{ item }">
+        <div class="flex items-center gap-2">
+          <UserAvatar
+            hide-tooltip
+            :user="item.creator"
+            light-style
+            class="bg-foundation"
+            no-bg
+          />
+          <span class="truncate text-body-xs text-foreground">
+            {{ item.creator?.name }}
+          </span>
+        </div>
+      </template>
+      <template #id="{ item }">
+        <div class="flex items-center text-foreground-2">
+          {{ item.id }}
+          <ClipboardIcon
+            class="w-4 h-4 ml-2 cursor-pointer hover:text-foreground transition"
+            @click="() => handleCopyText(item.id)"
+          />
+        </div>
+      </template>
+      <template #actions="{ item }">
+        <SettingsWorkspacesAutomationFunctionsTableRowActions
+          :workspace-function="item"
+          :permissions="item.permissions"
+        />
+      </template>
+    </LayoutTable>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { graphql } from '~/lib/common/generated/gql'
+import type { SettingsWorkspacesAutomationFunctions_WorkspaceFragment } from '~/lib/common/generated/gql/graphql'
+import { automateFunctionRoute } from '~/lib/common/helpers/route'
+import { ClipboardIcon } from '@heroicons/vue/24/outline'
+
+graphql(`
+  fragment SettingsWorkspacesAutomationFunctions_Workspace on Workspace {
+    id
+    automateFunctions(limit: 50, filter: { includeFeatured: false }) {
+      items {
+        id
```

https://www.notion.so/speckle/General-Web-Code-Review-Guide-1f2bf3dfea634460b3778ac3cf4aa1bb?source=copy_link#d40e40efc32f4e5ea67b68e848aadcd0

only some of the fields are directly requested by this component, a lot of them are actually used by child components which should then define their own fragments

---

I was having trouble getting the fragment to work in that one file because....it used the `satisfies` operator, something happened with parsing the .vue that failed to then find the fragment, and gqlgen failed silently. very fun debug